src/Microsoft.Health.SqlServer/Features/Schema/SchemaInitializer.cs

```diff
@@ -167,22 +163,24 @@ public static async Task<bool> DoesDatabaseExistAsync(SqlConnection sqlConnectio
             return false;
         }
 
+        [SuppressMessage("Security", "CA2100:Review SQL queries for security vulnerabilities", Justification = "Database name is validated before use.")]
         public static async Task<bool> CreateDatabaseAsync(SqlConnection connection, string databaseName, CancellationToken cancellationToken)
         {
-            using (var canCreateDatabaseCommand = new SqlCommand("SELECT count(*) FROM fn_my_permissions (NULL, 'DATABASE') WHERE permission_name = 'CREATE DATABASE'", connection))
+            if (!Identifier.IsValidDatabase(databaseName))
             {
-                if ((int)await canCreateDatabaseCommand.ExecuteScalarAsync(cancellationToken) > 0)
-                {
-                    using (var createDatabaseCommand = new SqlCommand($"CREATE DATABASE {databaseName}", connection))
-                    {
-                        await createDatabaseCommand.ExecuteNonQueryAsync(cancellationToken);
-                        return true;
-                    }
-                }
-                else
-                {
-                    return false;
-                }
+                throw new ArgumentException(string.Format(CultureInfo.InvariantCulture, Resources.InvalidDatabaseName, databaseName), nameof(databaseName));
+            }
+
+            using var canCreateDatabaseCommand = new SqlCommand("SELECT count(*) FROM fn_my_permissions (NULL, 'DATABASE') WHERE permission_name = 'CREATE DATABASE'", connection);
```

Comment:
There are many instances where the indentation has changed due to the newer `using` statements